lending/loan_management/doctype/loan/loan.py

```diff
@@ -72,7 +73,10 @@ def validate_cost_center(self):
 				frappe.throw(_("Cost center is mandatory for loans having rate of interest greater than 0"))
 
 	def set_cyclic_date(self):
-		if self.repayment_schedule_type == "Monthly as per cycle date":
+		if (
```

Comment:
Maybe rename to “~Monthly~ As per cycle date”?
You would check this even if frequency is quarterly or so…